android/pytorch_android/src/main/java/org/pytorch/Tensor.java

```diff
@@ -3,131 +3,167 @@
 import java.nio.Buffer;
 import java.nio.ByteBuffer;
 import java.nio.ByteOrder;
+import java.nio.DoubleBuffer;
 import java.nio.FloatBuffer;
 import java.nio.IntBuffer;
+import java.nio.LongBuffer;
 import java.util.Arrays;
 import java.util.Locale;
 
 public abstract class Tensor {
-  private static final int TYPE_CODE_BYTE = 1;
-  private static final int TYPE_CODE_INT32 = 2;
-  private static final int TYPE_CODE_FLOAT32 = 3;
+  public static final int DTYPE_BYTE = 1;
+  public static final int DTYPE_INT32 = 2;
+  public static final int DTYPE_FLOAT32 = 3;
+  public static final int DTYPE_LONG64 = 4;
+  public static final int DTYPE_DOUBLE64 = 5;
 
   private static final String ERROR_MSG_DATA_BUFFER_NOT_NULL = "Data buffer must be not null";
   private static final String ERROR_MSG_DATA_ARRAY_NOT_NULL = "Data array must be not null";
-  private static final String ERROR_MSG_DIMS_NOT_NULL = "Dims must be not null";
-  private static final String ERROR_MSG_DIMS_NOT_EMPTY = "Dims must be not empty";
-  private static final String ERROR_MSG_INDEX_NOT_NULL = "Index must be not null";
-  private static final String ERROR_MSG_DIMS_NON_NEGATIVE = "Dims must be non negative";
+  private static final String ERROR_MSG_SHAPE_NOT_NULL = "Dims must be not null";
+  private static final String ERROR_MSG_SHAPE_NOT_EMPTY = "Dims must be not empty";
```

Comment:
why are you doing these checks at the java level?  For one, they are wrong -- it's perfectly valid to have a zero-dimensional tensor.  Also, why can't you trust that the underlying C++ calls do the correct checks?

This might not apply to null because that's some java-specific thing you may not want to allow for API cleanliness, I don't have a judgement about that.